libafl_qemu/src/executor.rs

```diff
@@ -67,27 +76,61 @@ where
         })
     }
 
+    pub fn inner(&self) -> &InProcessExecutor<'a, H, I, OT, S> {
+        &self.inner
+    }
+
+    pub fn inner_mut(&mut self) -> &mut InProcessExecutor<'a, H, I, OT, S> {
+        &mut self.inner
+    }
+
     #[allow(clippy::unused_self)]
-    pub fn hook_edge_generation(&self, hook: fn(&mut S, u64, u64) -> Option<u32>) {
+    pub fn hook_edge_generation(&self, hook: fn(&mut S, src: u64, dest: u64) -> Option<u32>) {
         unsafe { GEN_EDGE_HOOK_PTR = hook as *const _ };
         emu::set_gen_edge_hook(gen_edge_hook_wrapper::<S>);
     }
 
     #[allow(clippy::unused_self)]
-    pub fn hook_edge_execution(&self, hook: extern "C" fn(u32)) {
+    pub fn hook_edge_execution(&self, hook: extern "C" fn(id: u32)) {
         emu::set_exec_edge_hook(hook);
     }
 
     #[allow(clippy::unused_self)]
-    pub fn hook_block_generation(&self, hook: fn(&mut S, u64) -> Option<u32>) {
+    pub fn hook_block_generation(&self, hook: fn(&mut S, addr: u64) -> Option<u32>) {
         unsafe { GEN_BLOCK_HOOK_PTR = hook as *const _ };
         emu::set_gen_block_hook(gen_block_hook_wrapper::<S>);
     }
 
     #[allow(clippy::unused_self)]
-    pub fn hook_block_execution(&self, hook: extern "C" fn(u64)) {
+    pub fn hook_block_execution(&self, hook: extern "C" fn(addr: u64)) {
         emu::set_exec_block_hook(hook);
     }
+
+    #[allow(clippy::unused_self)]
+    pub fn hook_cmp_generation(&self, hook: fn(&mut S, addr: u64, size: usize) -> Option<u32>) {
```

Comment:
Why not leave the `&self`? One less argument passed. Shouldn't matter fro LTO, but is there a benefit?

Reply:
I need self for the generic

Reply:
Makes sense thx